src/cs/Ssh/SshStream.cs

```diff
@@ -191,6 +191,8 @@ protected override void Dispose(bool disposing)
 			// Asynchronously close the channel, but don't wait for it.
 			_ = Channel.CloseAsync();
 
+			this.readSemaphore.TryRelease();
```

Comment:
Should we release it after setting `IsDisposed = true;` ? This way the waiting code can check `IsDisposed` reliably after waiting for `readSemaphore`.

Reply:
Updated.